### PR TITLE
refactor(batch): Use futures-async-stream to implement RowSeqScan executor

### PR DESCRIPTION
**Pull request by D2Lark**
## Implement RowSeqScan using Executor2 trait

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/1942

### CODE REVIEW

File: src/batch/src/executor2/row_seq_scan.rs [hunk 1]
```diff
@@ -99,58 +100,54 @@ impl BoxedExecutorBuilder for RowSeqScanExecutorBuilder {
...
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        self.do_execute()
+    }
+}
+
+impl<S: StateStore> RowSeqScanExecutor2<S> {
+    #[try_stream(boxed, ok = DataChunk, error = RwError)]
+    async fn do_execute(mut self: Box<Self>) {
+        if !self.should_ignore() {
+            self.iter = Some(self.table.iter(self.epoch).await?);
```
Comment (BugenZhao) [inline src/batch/src/executor2/row_seq_scan.rs:133]: We can make `iter` a local variable here.

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks for your contribution.

**Comment by liurenjie1024** [comment]
Forked in #2254

**Pull request closed without merging**